grails-app/conf/application.yml

```diff
@@ -177,7 +177,6 @@ grails:
     controllers:
         upload:
             maxFileSize: 2097152
-            maxRequestSize: 2097152
```

Comment:
I think it is not a good decision looking from the security context. Without a maximum request size, someone could send excessively large requests, overwhelming the server. This could lead to performance degradation or temporal server unavailability.

Reply:
Thanks for pointing that out, I had to check/test few things so...
The new `CommonsMultipartResolver` that I have implemented here receives a parameter `maxUploadSize` that checks the size of all of the uploaded files as a whole.

The standard multipart resolver (default one) receives parameters `maxFileSize` `maxRequestSize` and works a little differently.
the `maxFileSize` refers to the individual file size and `maxRequestSize` to the whole request size.

So in the end specifying  `maxFileSize: 2097152` and ` maxRequestSize: 2097152` is almost equivalent  to `maxUploadSize = 2097152`

Note that `maxRequestSize` only sets this limit on multipart request and not all of the API calls, so if we want to secure our API a bit better we need to look for additional property that would handle this validation.

Reply:
Also, there's probably a default value for maxRequestSize that is much smaller than what we've set it to. 

https://guides.grails.org/grails3/grails-upload-file/guide/index.html#:~:text=Grails%203's%20default%20file%20size,to%20allow%2025MB%20files%20uploads.